File: Python3/3392_count_subarrays_of_length_three_with_a_condition.py
# https://leetcode.cn/problems/count-subarrays-of-length-three-with-a-condition/

class Solution:
    def countSubarrays(self, nums):
        """
        Inputs:
            nums: list[int]
        Outputs:
            res: int
        """
        res = 0
        for i in range(1, len(nums)-1):
            if (nums[i-1] + nums[i+1]) * 2 == nums[i]:
                res += 1
        return res

File: Cpp/3392_count_subarrays_of_length_three_with_a_condition.cpp
// https://leetcode.cn/problems/count-subarrays-of-length-three-with-a-condition/

#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int countSubarrays(vector<int>& nums) {
        int res = 0;
        for (int i = 1; i < nums.size()-1; ++i) {
            res += (nums[i-1] + nums[i+1]) * 2 == nums[i];
        }
        return res;
    }
};

int main() {
    Solution sol;
    vector<int> nums = {1,2,1,4,1};
    cout << sol.countSubarrays(nums) << '\n';
    return 0;
}
